### PR TITLE
feat: reliable method of measuring accuracy

### PR DESCRIPTION
**Pull request by crashyspyro**
Closes https://github.com/sperezacuna/oracle-challenge-f2/issues/2.

Implements several ways of measuring F1-score

### CODE REVIEW

File: app/classifier/services.py [hunk 1]
```diff
@@ -53,13 +54,18 @@ def train(self, dataloaders, criterion, optimizer, scheduler):
...
               optimizer.step()
-          # statistics
+          # statistics  
           running_loss += loss.item() * inputs[0].size(0)
           running_corrects += torch.sum(preds == labels.data)
         if phase == 'training' and scheduler is not None:
           scheduler.step()
         epoch_loss = running_loss / len(dataloaders[phase].dataset)
-        epoch_acc = running_corrects.double() / len(dataloaders[phase].dataset)
+        epoch_acc = f1_score(labels.tolist(), preds.tolist(), average="macro")
```
Comment (sperezacuna) [inline app/classifier/services.py:63]: Big mistake here, comparing predictions of batch with labels of whole epoch (much larger). Must compare epoch-to-epoch, else it will not be a realistic measure.

Comment (sperezacuna) [inline app/classifier/services.py:63]: Big mistake here, only taking into account those labels and predictios corresponding to last batch. Must compare whole epochs, else the measure will be biased.

**Review by sperezacuna — CHANGES_REQUESTED** [review]
Biased [incomplete data taken into account] statistic.

File: app/classifier/services.py (hunk 1 above)
Comment (crashyspyro) [reply app/classifier/services.py]: Solved in https://github.com/sperezacuna/oracle-challenge-f2/pull/9/commits/fefc385cca7ca1a7fa603478cced16f2d138485f

**Review by sperezacuna — APPROVED** [review]
Fixed
